main.rb

```diff
@@ -0,0 +1,185 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    while i < length
+      temp.push(self[i]) if yield(self[i])
+      i += 1
+    end
+    temp
+  end
+
+  def my_all?
+    i = 0
+    while i < length
+      return false unless yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    i = 0
+    while i < length
+      return true if yield(self[i])
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?
+    i = 0
+    while i < length
+      return false if yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_count(filter = nil)
+    return length unless filter
+
+    count = 0
+    i = 0
+    while i < length
+      if block_given?
+        count += 1 if yield(self[i])
+      elsif self[i] == filter
+        count += 1
+      end
+      i += 0
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    return self unless block_given? || proc
+
+    temp = []
+    i = 0
+    if proc
+      while i < length
+        temp.push(proc.call(self[i]))
+        i += 1
+      end
+    else
+      while i < length
+        temp.push(yield(self[i]))
+        i += 1
+      end
+    end
+    temp
+  end
```

Comment:
* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1,2,3,4,5].my_map != [1,2,3,4,5].map
```